tests/tests/device.rs

```diff
@@ -91,3 +91,181 @@ async fn request_device_error_message() {
+
+            let texture_for_write = ctx.device.create_texture(&wgpu::TextureDescriptor {
+                label: None,
+                size: texture_extent,
+                mip_level_count: 2,
+                sample_count: 1,
+                dimension: wgpu::TextureDimension::D2,
+                format: wgpu::TextureFormat::Rg8Uint,
+                usage: wgpu::TextureUsages::COPY_DST,
+                view_formats: &[],
+            });
+
+            // Create some buffers.
+            let buffer_source = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+                label: None,
+                size: 256,
+                usage: wgpu::BufferUsages::MAP_WRITE | wgpu::BufferUsages::COPY_SRC,
+                mapped_at_creation: false,
+            });
+            let buffer_dest = ctx.device.create_buffer(&wgpu::BufferDescriptor {
+                label: None,
+                size: 256,
+                usage: wgpu::BufferUsages::MAP_READ | wgpu::BufferUsages::COPY_DST,
+                mapped_at_creation: false,
+            });
+
+            // Create some command encoders.
+            let mut encoder_for_clear = ctx
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+            let mut encoder_for_compute_pass = ctx
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+            let mut encoder_for_render_pass = ctx
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+            let mut encoder_for_buffer_buffer_copy = ctx
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+            let mut encoder_for_buffer_texture_copy = ctx
+                .device
+                .create_command_encoder(&wgpu::CommandEncoderDescriptor::default());
+
+            // Lose the device. This will cause all other requests to return some variation of a
+            // device invalid error.
+            ctx.device.lose();
+
+            // TODO: change these fail calls to check for the specific errors which indicate that
+            // the device is not valid.
+
+            // Creating a texture should fail.
+            fail(&ctx.device, || {
+                ctx.device.create_texture(&wgpu::TextureDescriptor {
+                    label: None,
+                    size: wgpu::Extent3d {
+                        width: 512,
+                        height: 512,
+                        depth_or_array_layers: 1,
+                    },
+                    mip_level_count: 2,
+                    sample_count: 1,
+                    dimension: wgpu::TextureDimension::D2,
+                    format: wgpu::TextureFormat::Rg8Uint,
+                    usage: wgpu::TextureUsages::COPY_SRC,
+                    view_formats: &[],
+                });
+            });
+
```

Comment:
missing a bunch of calls, e.g. create_buffer, clear_buffer

Reply:
I'll try to make it comprehensive.